script.js

```diff
@@ -1,2 +1,168 @@
+  movieDetails.style.display = "none";
+  movieDetails.style.color = "white";
+  movieDetails.innerHTML = `
+    <p><strong>Title:</strong> ${movie.title}</p>
+    <p><strong>Description:</strong> ${movie.description}</p>
+    <p><strong>Release Year:</strong> ${movie.movie_year}</p>
+    <p><strong>Director:</strong> ${movie.director}</p>
+    <p><strong>Actors:</strong> ${movie.actors.join(", ")}</p>
+  `;
+  return movieDetails;
+}
+
+function createCommentSection() {
+  const commentSection = document.createElement("div");
+  commentSection.innerHTML = "Comments: ";
+  commentSection.style.fontWeight = "bold";
+  commentSection.className = "comment-section";
+
+  const commentInput = document.createElement("input");
+  commentInput.type = "text";
+  commentInput.value = "";
+  commentInput.placeholder = "Add a Comment";
+  commentInput.style.color = "white";
+
+  const commentButton = document.createElement("button");
+  commentButton.className = "comment-button";
+  commentButton.innerHTML = "Submit";
+
+  const comments = document.createElement("div");
+  comments.className = "comments";
+  const commentUl = document.createElement("ul");
+
+  commentButton.addEventListener("click", () => {
+    if (commentInput.value) {
+      const comment = document.createElement("li");
+      comment.textContent = commentInput.value;
+      commentUl.appendChild(comment);
+      commentInput.value = "";
+    }
+    comments.appendChild(commentUl);
+  });
+
+  commentSection.appendChild(commentInput);
+  commentSection.appendChild(commentButton);
+  commentSection.appendChild(comments);
+
+  return commentSection;
+}
+
+function addEventListeners(readMoreBtn, movieDetails, stars) {
+  readMoreBtn.addEventListener("click", function () {
+    this.classList.toggle("active");
+    if (movieDetails.style.display === "block") {
+      movieDetails.style.display = "none";
+    } else {
+      movieDetails.style.display = "block";
+    }
+  });
+
+  const starClassActive = "rating__star fas fa-star";
+  const starClassInactive = "rating__star far fa-star";
+  stars.forEach((star, clickedStarIndex) => {
+    star.onclick = () => {
+      stars.forEach((star, starIndex) => {
+        if (starIndex <= clickedStarIndex) {
+          star.className = starClassActive;
+        } else {
+          star.className = starClassInactive;
+        }
+      });
+    };
+  });
+}
+
+function createMoviesPage() {
+  nowStreaming.innerHTML = "";
+  movies.forEach((movie) => {
+    const movieElement = createMovieCard(movie);
+    nowStreaming.appendChild(movieElement);
+  });
+}
+createMoviesPage();
+
+// JS 2
+const searchBtn = document.getElementById("search-icon");
+
+// function to search
+const inputField = document.getElementById("keyword-search");
+searchBtn.addEventListener("click", () => {
+  const inputStr = inputField.value.toLowerCase();
+  nowStreaming.innerHTML = "";
+  let found = false; // Flag to check if any movie is found
+
+  movies.forEach((movie) => {
```

Comment:
A `forEach` plus a `found` boolean can work fine. But you can probably simplify your code with one of the methods available on arrays. Have a look at https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array#instance_methods. I am sure you will find something that fits your needs.